subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -493,6 +504,20 @@ static int pa_sync_term_request(struct bt_conn *conn,
 	return err;
 }
 
+static bool bass_source_is_duplicate(uint32_t broadcast_id, uint8_t adv_sid, bt_addr_le_t addr)
+{
+	struct bass_recv_state_internal *state = bass_lookup_broadcast_id(broadcast_id);
+
+	if (state != NULL && state->state.adv_sid == adv_sid
+	    && state->state.addr.type == addr.type) {
+		LOG_DBG("recv_state already exists at src_id=0x%02X", state->state.src_id);
+
+		return true;
+	}
```

Comment:
It just hit me: Since the duplication check is on the joint value of `{broadcast_id, adv_sid, addr_typ}`, then we may have multiple receive states with the same broadcast ID, and thus `bass_lookup_broadcast_id` will only ever get the first one, and this function is not working 100% as intended. 

I believe that we need to move the loop from `bass_lookup_broadcast_id` in here, and compare all 3 values with each receive state.

Reply:
Yes, you are absolutely correct, good catch.